Supertext.Base.Core.Configuration/ConfigurationExtension.cs

```diff
@@ -22,13 +25,56 @@ public static class ConfigurationExtension
                 .OnActivating(setting => SettingActivating(setting, configuration));
         }
 
-        private static void SettingActivating(IActivatingEventArgs<object> obj, IConfiguration configuration)
+        private static void SettingActivating(IActivatingEventArgs<object> args, IConfiguration configuration)
         {
-            var section = obj.Instance.GetType().GetCustomAttribute<ConfigSectionAttribute>();
+            var section = args.Instance.GetType().GetCustomAttribute<ConfigSectionAttribute>();
             if (section != null)
             {
-                configuration.GetSection(section.SectionName).Bind(obj.Instance);
+                configuration.GetSection(section.SectionName).Bind(args.Instance);
             }
+
+            SetKeyVaultSecrets(args, configuration);
+        }
+
+        private static void SetKeyVaultSecrets(IActivatingEventArgs<object> args, IConfiguration configuration)
+        {
+            var configInstance = args.Instance;
+            var properties = configInstance.GetType().GetProperties();
+            foreach (var propertyInfo in properties)
+            {
+                var keyVaultSecret = propertyInfo.GetCustomAttributes<KeyVaultSecretAttribute>().SingleOrDefault();
+                if (keyVaultSecret != null)
+                {
+                    SetValueIfSome(propertyInfo, configInstance, configuration);
+                }
+            }
+        }
+
+        private static void SetValueIfSome(PropertyInfo propertyInfo, object configInstance, IConfiguration configuration)
+        {
+            var valueOption = GetSettingsValue(propertyInfo.Name, configuration);
+            if (valueOption.IsSome)
+            {
+                propertyInfo.SetValue(configInstance, Convert(valueOption.Value, propertyInfo.PropertyType));
+            }
+        }
+
+        private static object Convert(object value, Type targetType)
+        {
+            var tc = TypeDescriptor.GetConverter(targetType);
+            return tc.ConvertFrom(value);
+        }
+
+        private static Option<string> GetSettingsValue(string settingsKey, IConfiguration configuration)
+        {
+            var value = configuration.GetValue<string>(settingsKey);
+            if (value != null)
+            {
+                return Option<string>.Some(value);
+            }
+
+            Console.WriteLine($"Key {settingsKey} not available");
```

Comment:
Why writing to the console? What about logging?

Reply:
I didn't like to provide the logger as parameter in RegisterConfigurationsWithAppConfigValues(). 
What's your opinion?